samples/sample_common/include/sample_utils.h

```diff
@@ -495,6 +495,54 @@ mfxF64 CalculateFrameRate(mfxU32 nFrameRateExtN, mfxU32 nFrameRateExtD);
 mfxU16 GetFreeSurfaceIndex(mfxFrameSurface1* pSurfacesPool, mfxU16 nPoolSize);
 mfxU16 GetFreeSurface(mfxFrameSurface1* pSurfacesPool, mfxU16 nPoolSize);
 void FreeSurfacePool(mfxFrameSurface1* pSurfacesPool, mfxU16 nPoolSize);
+
+class mfxBitstreamWrapper : public mfxBitstream
+{
+public:
+    mfxBitstreamWrapper()
+        : mfxBitstream()
+    {}
+
+    mfxBitstreamWrapper(mfxU32 n_bytes)
+        : mfxBitstream()
+    {
+        Extend(n_bytes);
+    }
+
+    mfxBitstreamWrapper(const mfxBitstreamWrapper & bs_wrapper)
+    : mfxBitstream(bs_wrapper)
+    , m_data(bs_wrapper.m_data)
+    {
+        Data = m_data.data();
+    }
+
+    mfxBitstreamWrapper & operator= (const mfxBitstreamWrapper& bs_wrapper)
+    {
+        static_cast<mfxBitstream&>(*this) = static_cast<const mfxBitstream&>(bs_wrapper);
+        m_data = bs_wrapper.m_data;
+        Data   = m_data.data();
+
```

Comment:
W/ that implementation we need to have protection against self-assignment.

Reply:
Why? It wouldn't harm the data. I don't like protections like:
```
if (this != &bs_wrapper)
{
...
}
```
because it is optimization for rare case which would actually slow down each assignment